bugs/gh-267.txt

```diff
@@ -0,0 +1,795 @@
+++ value=/etc/netplan/50-cloud-init.yaml
++ '[' -z /etc/netplan/50-cloud-init.yaml -o /etc/netplan/50-cloud-init.yaml = UNSET ']'
++ for var in $@
++ eval 'value=$SAMPLE_CLOUDCONFIG > /dev/null'
+++ value=sethostname.yaml
++ '[' -z sethostname.yaml -o sethostname.yaml = UNSET ']'
++ for var in $@
++ eval 'value=$PRESERVE_INSTANCE > /dev/null'
+++ value=false
++ '[' -z false -o false = UNSET ']'
++ parse_args ./1.sh bionic
++ script=./1.sh
++ shift
++ '[' 1 = 0 ']'
++ '[' 1 -ne 0 ']'
++ case "$1" in
++ UBUNTU_RELEASE=bionic
++ NETCFG_FILE=/etc/netplan/50-cloud-init.yaml
++ shift
++ '[' 0 -ne 0 ']'
++ '[' -z bionic ']'
++ create_setup_proposed_script
++ '[' 0 -eq 0 ']'
++ cat
++ case "${UBUNTU_RELEASE}" in
++ ORIG_EXPECTED_LOG='Selected renderer '\''eni'\'' from priority list: None'
++ UPGRADE_EXPECTED_LOG='Selected renderer '\''eni'\'' from priority list: None'
++ ADDITIONAL_PKG=ifupdown
++ cat
++ echo '=== BEGIN ' bionic
+=== BEGIN  bionic
++ setup bionic sethostname.yaml
++ release=bionic
++ ud_file=sethostname.yaml
++ release=bionic
++ name=test-sru-bionic
++ lxc stop test-sru-bionic
++ lxc delete test-sru-bionic
+++ cat sethostname.yaml
++ lxc launch ubuntu-daily:bionic test-sru-bionic -c 'user.user-data=#cloud-config
+packages:
+ - ifupdown'
+Creating test-sru-bionic
+Starting test-sru-bionic
++ reproduce test-sru-bionic
++ name=test-sru-bionic
++ lxc exec test-sru-bionic -- cloud-init status --wait --long
+....................................................................
+status: done
+time: Wed, 24 Jun 2020 19:22:45 +0000
+detail:
+DataSourceNoCloud [seed=/var/lib/cloud/seed/nocloud-net][dsmode=net]
++ lxc exec test-sru-bionic rm /etc/network/interfaces.d/50-cloud-init.cfg
+rm: cannot remove '/etc/network/interfaces.d/50-cloud-init.cfg': No such file or directory
++ true
++ lxc exec test-sru-bionic rm /etc/netplan/50-cloud-init.yaml
++ lxc exec test-sru-bionic -- cloud-init clean --logs --reboot
++ true
++ sleep 3
++ lxc exec test-sru-bionic -- cloud-init status --wait --long
+..........
+status: done
+time: Wed, 24 Jun 2020 19:22:53 +0000
+detail:
+DataSourceNoCloud [seed=/var/lib/cloud/seed/nocloud-net][dsmode=net]
+++ lxc exec test-sru-bionic -- grep 'Selected renderer' /var/log/cloud-init.log
++ renderer_log='2020-06-24 19:22:49,422 - __init__.py[DEBUG]: Selected renderer '\''eni'\'' from priority list: None'
++ '[' '2020-06-24 19:22:49,422 - __init__.py[DEBUG]: ' '!=' '2020-06-24 19:22:49,422 - __init__.py[DEBUG]: Selected renderer '\''eni'\'' from priority list: None' ']'
++ echo 'SUCCESS: test-sru-bionic found Selected renderer '\''eni'\'' from priority list: None'
+SUCCESS: test-sru-bionic found Selected renderer 'eni' from priority list: None
++ lxc exec test-sru-bionic -- cat /etc/netplan/50-cloud-init.yaml
+cat: /etc/netplan/50-cloud-init.yaml: No such file or directory
++ echo 'FAILURE: /etc/netplan/50-cloud-init.yaml not found'
+FAILURE: /etc/netplan/50-cloud-init.yaml not found
```

Comment:
Is this failure expected ?

Reply:
@lucasmoura ahh expected in that I put the ct in the wrong place (I only wanted to cat the original NETCFG_FILE before we rebooted with both ifupdown and netplan both installed. After both pkgs are present we expect failure in bionic and eoan. I'm re-running the tests now with the cat in the appropriate place.